components/brave_extension/extension/brave_extension/manifest.json

```diff
@@ -53,6 +58,16 @@
       ],
       "run_at": "document_start",
       "all_frames": true
+    },
+    {
+      "matches": [
+        "*://*/*"
+      ],
+      "run_at": "document_start",
+      "js": [
+        "bundles/content-script.bundle.js",
+        "../../../../vendor/web-discovery-project/build/module-content-script.js"
```

Comment:
We need to understand performance implications before merging this (cc @atuchin-m)

Reply:
@iefremov Anything in particular I can help you check? This code should not have any overhead unless the user opts-in the Web Discovery Project. 

BTW @keur, shouldn't this be `build/core/content-script.bundle.js` instead?

Reply:
cool, so I just want to ensure that there is zero overhead with no opt-in, and that the overhead is okay if Web Discovery is actually enabled.

Reply:
When the WDP is enabled then I think overhead should be fairly low but judge for yourself: https://github.com/brave/web-discovery-project/blob/main/modules/web-discovery-project/sources/content.es#L159

It will inject a content script on any page open in the browser, and register a few event listener (e.g. scroll event, etc.), and will also extract some stuff from the DOM like metadata, and ads on some SERP like Google. Overall it looks fairly lightweight.

Reply:
The code is in that directory. I created symlinks for the files that needed a specific path to load correctly